### PR TITLE
feat: Added support for opening password-protected files

### LINKED ISSUE #349: How to open encrypted files by passing the password?
**Question**

I would like to know how to open encrypted files by passing the password?

### PR DESCRIPTION
**Pull request by dantehemerson**
**Description**

Add support to open protected files.

**Linked issues**

Please link any issues this pull request related to. Using the word "closes" before the
link will mean the issue is automatically closed by this Pull Request.
 
Fixes #349 

**Testing**

Please describe how your changes have been tested.

**Checklist**

- [ ] I have provided a good description of the change above
- [ ] I have added any necessary tests
- [ ] I have added all necessary type hints
- [ ] I have checked my linting (`docker-compose run --rm lint`)
- [ ] I have added/updated all necessary documentation
- [ ] I have updated `CHANGELOG.md`, following the format from
      [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).

### CODE REVIEW

**Comment by dantehemerson** [comment]
Sorry I'm not familiar with testing in python, so I cann't see if the test I added is passing. 

I tried to run `docker-compose run tests`, but I got this error:
```
(root) Additional property base is not allowed
```

**Comment by jstockwin** [comment]
Hey @dantehemerson,
Thanks for submitting this PR - I agree it's a useful feature that we should support.

By default GitHub doesn't run all the tests until I approve them (as a safety feature). I've just approved, so you'll see the output under the "continuous-integration" job soon!

File: py_pdf_parser/loaders.py [hunk 1]
```diff
@@ -69,7 +70,9 @@ def load(
     la_params = {**DEFAULT_LA_PARAMS, **la_params}
 
     pages: Dict[int, Page] = {}
-    for page in extract_pages(pdf_file, laparams=LAParams(**la_params)):
+    for page in extract_pages(
+        pdf_file, laparams=LAParams(**la_params), password=password
```
Comment (jstockwin) [inline py_pdf_parser/loaders.py:74]: Please could you add the new `password` parameter into the docstring of this function?

**Review by jstockwin — CHANGES_REQUESTED** [review]

File: tests/test_loaders.py [hunk 2]
```diff
@@ -11,6 +12,20 @@ def test_load_file(self):
...
+        )
+        document = load_file(file_path, password="p4ssword")
+        self.assertIsInstance(document, PDFDocument)
+
+    def test_load_protected_file_wrong_password(self):
+        file_path = os.path.join(
+            os.path.dirname(__file__), "data", "pdfs", "test_protected.pdf"
+        )
+        with self.assertRaises(PDFPasswordIncorrect):
+            load_file(file_path, password="wrong_password")
```
Comment (dantehemerson) [inline tests/test_loaders.py:27]: Add test for the case when the password provided is incorrect.

File: py_pdf_parser/loaders.py (hunk 1 above)
Comment (dantehemerson) [reply py_pdf_parser/loaders.py:76]: @jstockwin Done!

File: py_pdf_parser/loaders.py [hunk 3]
```diff
@@ -52,13 +53,15 @@ def load(
...
+        password (str, optional): Password for the encrypted PDF. Required if the
+            PDF is encrypted.
         la_params (dict): The layout parameters passed to PDF Miner for analysis. See
             the PDFMiner documentation here:
             https://pdfminersix.readthedocs.io/en/latest/reference/composable.html#laparams.
             Note that py_pdf_parser will re-order the elements it receives from PDFMiner
             so options relating to element ordering will have no effect.
-        pdf_file_path (str, optional): Passed to `PDFDocument`. See the documentation
-            for `PDFDocument`.
         kwargs: Passed to `PDFDocument`. See the documentation for `PDFDocument`.
```
Comment (dantehemerson) [inline py_pdf_parser/loaders.py:65]: Just reordered args in docs to match parameters order.

**Comment by dantehemerson** [comment]
@jstockwin I just fixed the comments and were able to run tests, all tests are passing.

**Review by jstockwin — APPROVED** [review]
Thanks @dantehemerson - really nice extra feature. I'll try and get that released now.

**Comment by jstockwin** [comment]
This is in release 0.12.0, which is now published to PyPI at https://pypi.org/project/py-pdf-parser/0.12.0/

**Pull request merged**